openmeter/ent/schema/subscription_entitlement.go

```diff
@@ -0,0 +1,47 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type SubscriptionEntitlement struct {
+	ent.Schema
+}
+
+func (SubscriptionEntitlement) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.IDMixin{},
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{},
+	}
+}
+
+func (SubscriptionEntitlement) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("entitlement_id").NotEmpty().Immutable(),
+		field.String("subscription_id").NotEmpty().Immutable(),
+		field.String("subscription_phase_key").NotEmpty().Immutable(),
+		field.String("subscription_item_key").NotEmpty().Immutable(),
```

Comment:
any reason we don't reference `subscription_item` by ID?

Reply:
there were, but currently there isn't